game-core/src/main/java/games/strategy/debug/ErrorMessageFormatter.java

```diff
@@ -0,0 +1,179 @@
+   *      LogRecord.getMessage() == null && LogRecord.getThrown().getMessage() == null
+   * }
+   * Return:
+   * * {exception simple name}
+   * <br/>
+   * (2) uncaught exception with a message
+   * {@code
+   *         throw new NullPointerException("message") =>
+   *              // log record message will be populated with the exception message
+   *              LogRecord.getMessage().equals(LogRecord.getThrown().getMessage())}
+   * }
+   * <br/>
+   * Return:
+   * * {exception simple name} - {exception message}
+   * <br/>
+   * (3) logging an error message
+   * {@code
+   *         log.severe("message") =>
+   *              LogRecord.getMessage() != null && LogRecord.getThrown() == null
+   * }
+   * <br/>
+   * Return:
+   * * {log message}
+   * <br/>
+   * (4) logging an error message with exception that has no message</li>
+   * {@code
+   *        log.log(Level.SEVERE, "message", new NullPointerException()) =>
+   *              LogRecord.getMessage() != null && LogRecord.getThrown().getMessage() == null
+   * }
+   * Return:
+   * * * {log message}
+   * * *
+   * * * {exception simple name}
+   * <br/>
+   * (5) logging an error message with exception that has a message
+   * {@code
+   *        log.log(Level.SEVERE, "log-message", new NullPointerException("exception message")) =>
+   *              LogRecord.getMessage() != null
+   *                 && LogRecord.getThrown() != null
+   *                 && !LogRecord.getMessage().equals(LogRecord.getThrown().getMessage())
+   * }
+   * <br/>
+   * Return:
+   * * {log message}
+   * *
+   * * {exception simple name} - {exception message}
+   */
+  private static String format(final LogRecord logRecord) {
+    if (logRecord.getThrown() == null) {
+      return logMessageOnly(logRecord);
+    }
+
+    if (logRecord.getMessage() != null
+        && logRecord.getThrown() != null
+        && logRecord.getThrown().getMessage() != null
+        && !logRecord.getMessage().equals(logRecord.getThrown().getMessage())) {
+      return logMessageAndExceptionMessage(logRecord);
+    }
+
+    if (logRecord.getMessage() != null
+        && logRecord.getThrown() != null
+        && logRecord.getThrown().getMessage() == null) {
+      return logMessageAndExceptionWithoutMessage(logRecord);
+    }
+
+    if (logRecord.getMessage() == null
+        && logRecord.getThrown() != null
+        && logRecord.getThrown().getMessage() == null) {
+      return exceptionOnlyWithOutMessage(logRecord);
+    }
+
+    if (logRecord.getMessage() != null
+        && logRecord.getThrown() != null
+        && logRecord.getThrown().getMessage() != null
+        && logRecord.getMessage().equals(logRecord.getThrown().getMessage())) {
+      return exceptionOnlyWithMessage(logRecord);
+    }
+
+
+    throw new IllegalStateException("Unhandled: " + logRecord.getMessage() + ", exception: " + logRecord.getThrown());
+  }
+
+  /*
+   * <pre>
+   * Error: {log message}
+   * </pre>
+   */
```

Comment:
Those comments are odd, they should be converted into valid Javadocs.

Reply:
As javadocs then checkstyle kicks in and they need leading sentences with a period. The formatter as well which makes the `<pre>` blocks necessary, which again makes us need the leading sentences.

Reply:
I already assumed that, that's why I used the term "convert" as in "change it so it works".
I think at some point someone will look at those comments and wonder why those comments aren't javadocs.

Reply:
The checkstyle requirements of javadoc are a point of friction, it should not be this hard to document the code. I was moving to format the html/javadoc so this looked better, I do agree it was odd, but found an example that was out of date. The out of date example is good evidence the comments are no longer useful, they have been removed: 4b3836e